packages/expo-constants/src/ExponentConstants.web.ts

```diff
@@ -31,28 +37,34 @@ export default {
     return location.origin + location.pathname;
   },
   get expoRuntimeVersion(): string | null {
-    console.warn(`ExponentConstants.expoRuntimeVersion: is unimplemented on this platform.`);
     return null;
   },
   get deviceName(): string | null {
-    return null;
+    const { browser, engine, os: OS } = parser.getResult();
+
+    return `${browser.name || engine.name || OS.name}`;
   },
   get systemFonts(): string[] {
+    // TODO: Bacon: Maybe possible.
     return [];
   },
   get statusBarHeight(): number {
     return 0;
   },
   get deviceYearClass(): string | null {
-    console.warn(`ExponentConstants.deviceYearClass: is unimplemented on this platform.`);
+    // TODO: Bacon: The android version isn't very accurate either, maybe we could try and guess this value.
+    console.log(`ExponentConstants.deviceYearClass: is unimplemented on web.`);
     return null;
   },
   get manifest(): { [manifestKey: string]: any } {
-    /* TODO: Bacon: Populate */
-
-    return {};
+    let manifest;
+    // Bacon: Get manifest from webpack.config.js
+    if (process) {
```

Comment:
IIRC the Webpack plugin searches for `process.env.XX` rather than defining `process` -- just return `process.env.APP_MANIFEST || {}` here